lib/pages/dashboard/dashboard_club_teams.dart

```diff
@@ -38,59 +36,70 @@ class _DashboardClubTeamsState extends State<DashboardClubTeams> with SingleTick
-      setState(() {
-        _teams = teams;
-        _pageController.jumpTo(0);
-      });
-    });
+    if (widget.clubCode != oldWidget.clubCode) {
+      _clubTeamsBloc.add(ClubTeamsLoadEvent(clubCode: widget.clubCode));
+    }
   }
 
   @override
   Widget build(BuildContext context) {
-    return Column(
-      children: <Widget>[
-        Container(
-          height: widget.cardHeight,
-          child: PageView.builder(
-            physics: const BouncingScrollPhysics(),
-            controller: _pageController,
-            itemCount: _teams.length,
-            itemBuilder: (context, index) {
-              return TeamCard(
-                currentlyDisplayed: _currentIndex == index,
-                team: _teams[index],
-                distance: _pageController.page - index,
-              );
-            },
-          )
-        ),
-        if (_teams.length > 1)
-          Padding(
-            padding: EdgeInsets.only(top: 16.0),
-            child: SmoothPageIndicator(
-              controller: _pageController,
-              count: _teams.length,
-              effect: WormEffect(
-                dotHeight: 8,
-                dotWidth: 8,
-                dotColor: Theme.of(context).cardTheme.color,
-                activeDotColor: Theme.of(context).accentColor,
-              )
-            ),
-          )
-      ]
+    return BlocBuilder<ClubTeamsBloc, ClubTeamsState>(
+      bloc: _clubTeamsBloc,
+      builder: (context, state) {
+        if (state is ClubTeamsLoaded) {
+          return Column(
+              children: <Widget>[
+                Container(
+                    height: widget.cardHeight,
+                    child: PageView.builder(
+                      physics: const BouncingScrollPhysics(),
+                      controller: _pageController,
+                      itemCount: state.teams.length,
+                      itemBuilder: (context, index) {
+                        return TeamCard(
+                          currentlyDisplayed: _currentIndex == index,
+                          team: state.teams[index],
+                          distance: (_currentIndex - index).toDouble(),
+                        );
+                      },
+                    )
+                ),
+                if (state.teams.length > 1)
+                  Padding(
+                    padding: EdgeInsets.only(top: 16.0),
+                    child: SmoothPageIndicator(
+                        controller: _pageController,
+                        count: state.teams.length,
+                        effect: WormEffect(
+                          dotHeight: 8,
+                          dotWidth: 8,
+                          dotColor: Theme.of(context).cardTheme.color,
+                          activeDotColor: Theme.of(context).accentColor,
+                        )
+                    ),
+                  )
+              ]
+          );
+        } else {
+          return Container(height: widget.cardHeight);
```

Comment:
Is there any `ClubTeamsLoading` state? To display a loader

Reply:
Yes ! I just added a loader. I'm not sure it renders well however, when teams are loaded quickly it makes too much "transitions" in a small period of time on the screen.

More generally, I think it would be nice to display a loader only when the data takes a long time (threshold to define) to fetch. For example, a loader displayed for only 0.5s is not nice.